fontir/src/stateset.rs

```diff
@@ -345,21 +400,25 @@ mod tests {
     fn read_write_toml() {
         let temp_dir = tempdir().unwrap();
 
-        let (_, _, fs) = one_changed_file_one_not(&temp_dir);
+        let (_, _, mut fs) = one_changed_file_one_not(&temp_dir);
+        fs.track_slice(Path::new("file.glyphs/glyph/glyph_name"), "Hi World!")
+            .unwrap();
 
         let toml = toml::ser::to_string_pretty(&fs).unwrap();
-        let restored: FileStateSet = toml::from_str(&toml).expect(&toml);
+        let restored: StateSet = toml::from_str(&toml).expect(&toml);
         assert_eq!(fs, restored);
     }
 
     #[test]
     fn read_write_bincode() {
         let temp_dir = tempdir().unwrap();
 
-        let (_, _, fs) = one_changed_file_one_not(&temp_dir);
+        let (_, _, mut fs) = one_changed_file_one_not(&temp_dir);
+        fs.track_slice(Path::new("file.glyphs/glyph/glyph_name"), "Hi World!")
```

Comment:
what I think would be clearer and more explicit as type of the key in a SliceSet is a tuple/struct where one field is an actual Path or PathBuf (i.e. the path to the file containing the slice) and a second field contains a vector of strings defining the "path" components to identify the given slice

from `Path::new("file.glyphs/glyph/glyph_name"` to `(Path::new("file.glyphs"), vec!["glyph", "glyph_name"])`

Reply:
or as a struct:

```rust
let glyph_slice_path = { path: Path::new("file.glyphs"), keys: vec!["glyph", "glyph_name"] };
```

Reply:
My thought was that path captures that the user is intended to use a "pathlike" construct as the key, e.g. even for slices you probably want a structured key. I suppose I could hide that, and track_slice could input a String.

Reply:
track_slice now takes &str identifiers